server/tserver/src/main/java/org/apache/accumulo/tserver/log/TabletServerLogger.java

```diff
@@ -458,6 +458,26 @@ private void write(final Collection<CommitSession> sessions, boolean mincFinish,
       // the logs haven't changed.
       final int finalCurrent = currentLogId;
       if (!success) {
```

Comment:
There seem to be two cases were success is false.  One is when exceptions happen and the other is a race condition where the [logs change during a write](https://github.com/apache/accumulo/blob/78cbdbe5560272028b93401b9f2f4f792c625798/server/tserver/src/main/java/org/apache/accumulo/tserver/log/TabletServerLogger.java#L432-L440).  For the case where logs change that could be somewhat normal and may create a lot of zookeeper load.  Like if there were 100 threads writing mutations and the logs changes while they are doing that then all of those threads would go do these zookeeper reads.   

Could avoid checking on this race condition w/ another boolean that is only set when an exception happens.


```java
if(!success){
  if(sawException && (tserver.getLock() == null || !tserver.getLock().verifyLockAtSource())) {
     ....
  }
}

Reply:
boolean added in 77492bf